torch/_functorch/aot_autograd.py

```diff
@@ -436,11 +436,21 @@ class ViewAndMutationMeta:
     # another user output (in both cases, we won't redundantly append bases to the end of the graph)
     num_intermediate_bases: int
 
+    # For inference only: instructs us to keep data-only input mutations directly in the graph
+    keep_input_mutations: int
+
     def __post_init__(self):
         # pre-compute the indices of the inputs that are mutated
-        mutated_inp_indices = [
-            i for i, m in enumerate(self.input_info) if m.mutates_data or m.mutates_metadata
-        ]
+        if self.keep_input_mutations:
+            # When keep_input_mutations is set, we don't need to worry about our epilogue
+            # handling data-only mutations, because we keep them directly in the graph.
+            mutated_inp_indices = [
+                i for i, m in enumerate(self.input_info) if m.mutates_metadata
+            ]
+        else:
+            mutated_inp_indices = [
+                i for i, m in enumerate(self.input_info) if m.mutates_data or m.mutates_metadata
+            ]
```

Comment:
Could this just be:

```suggestion
          # When keep_input_mutations is set, we don't need to worry about our epilogue
          # handling data-only mutations, because we keep them directly in the graph.
          mutated_inp_indices = [
              i for i, m in enumerate(self.input_info) if m.mutates_metadata or (self.keep_input_mutations and  m.mutates_data)
          ]
```

Reply:
good catch, thanks - (it should actually be `if m.mutates_metadata or (not self.keep_input_mutations and  m.mutates_data)` -  "keep_input_mutations" means "keep input mutations in the *graph*, so we should *not* include them in the list of mutated input indices to handle in the epilogue")